### PR TITLE
Final Milestone : React Capstone Project

### PR DESCRIPTION
**Pull request by piesieomane**
In this PR the following list below was implemented:
- Continents Page
 Selected the Country restful API.
 Rendered the world map alongside the six continents which can be filtered to one.
- Countries Page
 An Action is sent to the API to request for all countries in a particular continent
 When the user clicks on the "Back" button (<), the user navigates to the home page.
- Details page
 This page shows the various statistics about the selected country.
  When the user clicks on the "Back" button (<), the user navigates to the home page.
- Testing requirements
Created unit tests for pure functions of the react components

- Technical requirements
  -- The project is a single-page application (SPA) built with React and Redux.
  -- The data retrieved from the API is stored in the Redux store.
  -- Filtered the data you retrieve from the API using a Filter stateless component.
  -- Every page (the main page and the pages for each item) has a unique route within the SPA.
  -- The project is deployed and accessible online.
  --Here's the link to my Loom Video Presentation.
  --(https://www.loom.com/share/d1ba867605ec4c8bbaec12182d71aecb4)

### CODE REVIEW

File: src/_tests_/continents.test.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    <React.StrictMode>
+      <Provider store={store}>
+        <BrowserRouter>
+          <Continents />
+        </BrowserRouter>
+      </Provider>
+    </React.StrictMode>
+  ).toJSON();
+  expect(tree).toMatchSnapshot();
+});
```
Comment (medaminedev66) [inline src/_tests_/continents.test.js:19]: - [ ] Following the project requirements, you should use also React Testing Library to simulate user interaction. Check out this [link ](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/articles/testing_react_with_react_testing_library.md)to learn how to do that.

File: src/_tests_/countries.test.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import countries from '../redux/countries/countries';
+
+it('should return the initial state', () => {
+  expect(countries(undefined, {})).toEqual([]);
+});
```
Comment (medaminedev66) [inline src/_tests_/countries.test.js]: - [ ] Following the project requirements, you should add unit tests for some pure functions in Redux code.

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @piesieomane  😃, :wave:, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- The repo looks professional.
- You have a descriptive PR.
- Your code organization was really good.
- Your web app works as expected and runs without errors.
- The project is deployed properly.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by t-yanick — APPROVED** [review]
Hi @piesieomane, 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**